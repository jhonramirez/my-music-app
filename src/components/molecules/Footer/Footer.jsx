import { useState } from "react";
import ReactPlayer from "react-player";
import { previousSong, nextSong, playSong, pauseSong } from "../../../svg";
import { Button } from "../..";
import "./Footer.scss";

export const Footer = ({ urlTrack}) => {
  const [playing, setPlaying] = useState(false);
  return (
    <>
      <div className="footer">
        <div className="footer__soundbar" />
        <div>
          <ReactPlayer url={urlTrack} playing={playing} className="footer__player" />
          <Button label={previousSong} />
          <Button
            label={playing ? pauseSong : playSong}
            onClick={() => {
              setPlaying(!playing);
            }}
          />
          <Button label={nextSong} />
        </div>
      </div>
    </>
  );
};
