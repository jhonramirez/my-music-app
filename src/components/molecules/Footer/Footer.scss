.footer {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-width: 350px; 
  margin-left: -125px; 
  background-color: rgb(17, 17, 16);
  border-radius: 20px; 
  padding: 10px; 
  position: fixed; 
  left: 50%; 
  bottom: 0px; 
  &__soundbar {
    min-width: 90%;
    background-color: rgba(248, 248, 248, 0.781);
    height: 4px;
    border-radius: 30px;
    cursor: pointer;
  }
  &__player{
    display: none;
  }
}

@media only screen and (max-device-width: 750px) {
  .footer {
    padding: 0.5rem;
    min-width: 250px; 
  }
}
