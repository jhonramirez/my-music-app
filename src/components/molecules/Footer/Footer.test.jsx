import { fireEvent, render, screen } from "@testing-library/react";
import { Footer } from "./Footer";
import "@testing-library/jest-dom";

describe("Pruebas en <Footer />", () => {
  
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<Footer urlTrack={"proof"} />);
    expect(container).toMatchSnapshot();
  });

  test("Se evalua que se ejecute el onClick", () => {
    render(<Footer urlTrack={"proof"} />);
    const buttonElement = screen.getAllByRole("button")[1];
    fireEvent.click(buttonElement);
  });
});