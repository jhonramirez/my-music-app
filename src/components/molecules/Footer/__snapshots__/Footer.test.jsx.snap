// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Footer /> Debe hacer match con el snapshot 1`] = `
<div>
  <section
    class="footer"
    data-testid="footer"
  >
    <article
      class="footer__soundbar"
    />
    <article>
      <div
        class="footer__player"
        style="width: 640px; height: 360px;"
      />
      <button
        class="btn btn__black"
        data-testid="button"
      >
        <svg
          class=""
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M13 2.5L5 7.119V3H3v10h2V8.881l8 4.619z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="btn btn__black"
        data-testid="button"
      >
        <svg
          class="bi bi-play"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
           
          <path
            d="M10.804 8 5 4.633v6.734L10.804 8zm.792-.696a.802.802 0 0 1 0 1.392l-6.363 3.692C4.713 12.69 4 12.345 4 11.692V4.308c0-.653.713-.998 1.233-.696l6.363 3.692z"
          />
           
        </svg>
      </button>
      <button
        class="btn btn__black"
        data-testid="button"
      >
        <svg
          class=""
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M11 3v4.119L3 2.5v11l8-4.619V13h2V3z"
            fill="currentColor"
          />
        </svg>
      </button>
    </article>
  </section>
</div>
`;
